test_node.py

```diff
@@ -1,19 +1,133 @@
+from dataclasses import dataclass
+from os.path import basename
+
+# from typing import List
+
 import pytest
-import testinfra
+
+
+from conftest import restrict_to_version
+
 
 # Container fixture contains the black magic to run command on all the different kind of nodes
 # per language.
 # If you need to run a test for a single version, create your own fixture
 # You should think of reusing the `container` fixture from yours.
 
+
+@dataclass(frozen=True)
+class NpmPackageTest:
+    # node_versions: List[int] = field(default_factory=list)
+    build_command: str = ""
+    repository_url: str = ""
+
+    @property
+    def repo_name(self) -> str:
+        return basename(self.repository_url.replace(".git", ""))
+
+    @property
+    def test_command(self) -> str:
+        return f"""git clone {self.repository_url} &&
+cd {self.repo_name} &&
+{self.build_command}
+"""
+
+    def __str__(self) -> str:
+        return self.repo_name
```

Comment:
that will be a long name, I would extract just the last word of the the url instead (basename).

Reply:
No, `repo_name` is e.g. `react` or `lodash`.

Reply:
I see now it is using basename, good.